cpp/test/neighbors/ann_ivf_pq.cuh

```diff
@@ -962,6 +1040,19 @@ inline auto special_cases() -> test_cases_t
     x.search_params.n_probes     = 100;
   });
 
+  ADD_CASE({
+    x.num_db_vecs                = 4335;
+    x.dim                        = 4;
+    x.num_queries                = 100000;
+    x.k                          = 12;
+    x.index_params.metric        = distance::DistanceType::L2Expanded;
+    x.index_params.codebook_kind = ivf_pq::codebook_gen::PER_SUBSPACE;
```

Comment:
What happens if the codebook kind is per cluster? Can we add a few more sets of params here just as a sanity check?

Reply:
FAISS does not support PER_CLUSTER. In the FAISS current integration work [only PER_SUBSPACE](https://github.com/facebookresearch/faiss/pull/3044/files#diff-c0dd2ef12b13e15f7ce9a5f0abf3bca51445c8965c62ebd4a75305a11d0bcc4aR139) is used to be consistent with faiss cpu.

Reply:
I get that- but this is a RAFT test not a FAISS test and I don't see `PER_SUBSPACE` anywhere else in this PR. Shouldn't both work? Are you explicitly asserting PER_SUBSPACE somewhere in the helpers?

Reply:
Our public APIs are exposed to all the public. We shouldn't be cherry-picking users that we're catering towards. If this is generalized enough to be part of the public API, we need to make sure we're testing as such.

Reply:
Added the PER_CLUSTER test case.